src/parser.js

```diff
@@ -1319,9 +1319,18 @@ export class Parser extends Tokenizer {
   parseNewExpression() {
     let startLocation = this.getLocation();
     this.expect(TokenType.NEW);
+    if (this.inFunctionBody && this.eat(TokenType.PERIOD)) {
+      let ident = this.expect(TokenType.IDENTIFIER);
+      if (ident.value !== "target" || ident.slice.end - ident.slice.start !== 6) {
```

Comment:
The size test is not necessary:

> All interpretations of IdentifierName within this specification are based upon their actual code points regardless of whether or not an escape sequence was used to contribute any particular code point.
https://people.mozilla.org/~jorendorff/es6-draft.html#sec-names-and-keywords 

Exactly why we don't need to check the size of other contextual keywords.